### PR TITLE
[FIX] ui5Framework.mergeTrees: Do not process the same project multiple times

### PR DESCRIPTION
**Pull request by RandomByte**
Since projects are deduped by the projectPreprocessor, repeated
processing of a project that appears multiple times in the project tree
is done on the same object. This leads to issues like the
isFrameworkProject detecting previously added framework libraries as ...
"already added UI5 framework libraries", thus producing incorrect log
messages

The eventual result in projects where a reuse library defining UI5 dependencies has multiple occurrences in the tree is excessive logging like this:
```sh
info normalizer:translators:ui5Framework If project my.pony.collection contains a package.json in which it defines a dependency to the UI5 framework library @openui5/sap.f, this dependency should be removed.
info normalizer:translators:ui5Framework If project my.pony.collection contains a package.json in which it defines a dependency to the UI5 framework library @openui5/sap.m, this dependency should be removed.
info normalizer:translators:ui5Framework If project my.pony.collection contains a package.json in which it defines a dependency to the UI5 framework library @openui5/sap.ui.core, this dependency should be removed.
info normalizer:translators:ui5Framework If project my.pony.collection contains a package.json in which it defines a dependency to the UI5 framework library @openui5/sap.ui.layout, this dependency should be removed.
info normalizer:translators:ui5Framework If project my.pony.collection contains a package.json in which it defines a dependency to the UI5 framework library @sapui5/sap.viz, this dependency should be removed.
info normalizer:translators:ui5Framework If project my.pony.collection contains a package.json in which it defines a dependency to the UI5 framework library @openui5/sap.f, this dependency should be removed.
info normalizer:translators:ui5Framework If project my.pony.collection contains a package.json in which it defines a dependency to the UI5 framework library @openui5/sap.m, this dependency should be removed.
info normalizer:translators:ui5Framework If project my.pony.collection contains a package.json in which it defines a dependency to the UI5 framework library @openui5/sap.ui.core, this dependency should be removed.
info normalizer:translators:ui5Framework If project my.pony.collection contains a package.json in which it defines a dependency to the UI5 framework library @openui5/sap.ui.layout, this dependency should be removed.
info normalizer:translators:ui5Framework If project my.pony.collection contains a package.json in which it defines a dependency to the UI5 framework library @sapui5/sap.viz, this dependency should be removed.
info normalizer:translators:ui5Framework If project my.pony.superlib contains a package.json in which it defines a dependency to the UI5 framework library @openui5/sap.f, this dependency should be removed.
info normalizer:translators:ui5Framework If project my.pony.superlib contains a package.json in which it defines a dependency to the UI5 framework library @openui5/sap.m, this dependency should be removed.
info normalizer:translators:ui5Framework If project my.pony.superlib contains a package.json in which it defines a dependency to the UI5 framework library @openui5/sap.ui.core, this dependency should be removed.
info normalizer:translators:ui5Framework If project my.pony.superlib contains a package.json in which it defines a dependency to the UI5 framework library @openui5/sap.ui.layout, this dependency should be removed.
info normalizer:translators:ui5Framework If project my.pony.superlib contains a package.json in which it defines a dependency to the UI5 framework library @openui5/sap.ui.unified, this dependency should be removed.
info normalizer:translators:ui5Framework If project my.pony.superlib contains a package.json in which it defines a dependency to the UI5 framework library @sapui5/sap.viz, this dependency should be removed.
info normalizer:translators:ui5Framework If project my.pony.superlib contains a package.json in which it defines a dependency to the UI5 framework library @openui5/themelib_sap_fiori_3, this dependency should be removed.
info normalizer:translators:ui5Framework If project my.pony.collection contains a package.json in which it defines a dependency to the UI5 framework library @openui5/sap.f, this dependency should be removed.
info normalizer:translators:ui5Framework If project my.pony.collection contains a package.json in which it defines a dependency to the UI5 framework library @openui5/sap.m, this dependency should be removed.
info normalizer:translators:ui5Framework If project my.pony.collection contains a package.json in which it defines a dependency to the UI5 framework library @openui5/sap.ui.core, this dependency should be removed.
info normalizer:translators:ui5Framework If project my.pony.collection contains a package.json in which it defines a dependency to the UI5 framework library @openui5/sap.ui.layout, this dependency should be removed.
info normalizer:translators:ui5Framework If project my.pony.collection contains a package.json in which it defines a dependency to the UI5 framework library @sapui5/sap.viz, this dependency should be removed.
info normalizer:translators:ui5Framework If project my.pony.collection contains a package.json in which it defines a dependency to the UI5 framework library @openui5/sap.f, this dependency should be removed.
info normalizer:translators:ui5Framework If project my.pony.collection contains a package.json in which it defines a dependency to the UI5 framework library @openui5/sap.m, this dependency should be removed.
info normalizer:translators:ui5Framework If project my.pony.collection contains a package.json in which it defines a dependency to the UI5 framework library @openui5/sap.ui.core, this dependency should be removed.
info normalizer:translators:ui5Framework If project my.pony.collection contains a package.json in which it defines a dependency to the UI5 framework library @openui5/sap.ui.layout, this dependency should be removed.
info normalizer:translators:ui5Framework If project my.pony.collection contains a package.json in which it defines a dependency to the UI5 framework library @sapui5/sap.viz, this dependency should be removed.
```

### CODE REVIEW

File: test/lib/translators/ui5Framework.integration.js [hunk 1]
```diff
@@ -143,28 +147,31 @@ function defineTest(testName, {
 		};
 	}
 
-	test.serial(`${frameworkName}: ${verbose ? "(verbose) " : ""}${testName}`, async (t) => {
+	test.serial.only(`${frameworkName}: ${verbose ? "(verbose) " : ""}${testName}`, async (t) => {
```
Comment (codeworrior) [inline test/lib/translators/ui5Framework.integration.js:150]: forgotten?

File: lib/translators/ui5Framework.js [hunk 2]
```diff
@@ -212,8 +212,18 @@ module.exports = {
...
 		const queue = [projectTree];
+		const processedProjects = [];
 		while (queue.length) {
 			const project = queue.shift();
+			if (processedProjects.includes(project.id)) {
+				// projectTree must be duplicate free. A second occurrence of the same project
+				//	is always the same object. Therefore a single processing needs to be ensured.
+				// Otherwise the isFrameworkProject check would detect framework dependencies added
+				//	at an earlier processing of the project and yield incorrect logging.
+				log.verbose(`Project ${project.metadata.name} (${project.id}) has already peen processed`);
```
Comment (matz3) [inline lib/translators/ui5Framework.js:223]:
```suggestion
				log.verbose(`Project ${project.metadata.name} (${project.id}) has already been processed`);
```

File: test/lib/translators/ui5Framework.integration.js (hunk 1 above)
Comment (matz3) [inline test/lib/translators/ui5Framework.integration.js:150]:
```suggestion
	test.serial(`${frameworkName}: ${verbose ? "(verbose) " : ""}${testName}`, async (t) => {
```
I think so

Comment (RandomByte) [reply test/lib/translators/ui5Framework.integration.js]: Done

File: lib/translators/ui5Framework.js (hunk 2 above)
Comment (RandomByte) [reply lib/translators/ui5Framework.js]: Done

File: test/lib/translators/ui5Framework.integration.js (hunk 1 above)
Comment (RandomByte) [reply test/lib/translators/ui5Framework.integration.js]: Done

**Pull request merged**